collect_app/src/main/java/org/odk/collect/android/preferences/PreferencesFragment.java

```diff
@@ -129,6 +133,39 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
         }
     }
 
+    private void initLanguagePrefs() {
+        final ListPreference pref = (ListPreference) findPreference(KEY_LANGUAGE);
+        final LocaleHelper localeHelper = new LocaleHelper();
+        TreeMap<String, String> languageList = localeHelper.getEntryListValues();
+        pref.setEntryValues(languageList.values().toArray(new String[0]));
```

Comment:
I think you want to pass the size of the array in the toArray() here. If you dig into the toArray() implementation, it will use the array that you passed in to the method if the size is greater or equal to the element to be converted to array or create a new array if you passed in a smaller size.

Since you're passing array size 0 here, it will create a new String array object. So you will have 2 instances of String array during this process. I know it the memory footprint probably pretty small, but we don't want to create unnecessary objects when we don't need to.